ietf/iesg/tests.py

```diff
@@ -500,12 +499,46 @@ def test_agenda_documents_txt(self):
     def test_agenda_documents(self):
         url = urlreverse("ietf.iesg.views.agenda_documents")
         r = self.client.get(url)
+
+        dates = list(TelechatDate.objects.order_by('date').values_list("date", flat=True)[:4])
+        docs_by_date = dict((d, []) for d in dates)
+        docs = Document.objects.distinct()
+        for doc in docs:
+            d = doc.telechat_date()
+            if d in docs_by_date:
+                # Set some states that affect the returned ad_pages_left_to_ballot_on
+                doc.set_state(State.objects.get(used=True, type="draft-iesg", slug="pub"))
+
         self.assertEqual(r.status_code, 200)
 
         for k, d in self.telechat_docs.items():
             self.assertContains(r, d.name, msg_prefix="%s '%s' not in response" % (k, d.name, ))
-            self.assertContains(r, d.title, msg_prefix="%s '%s' title not in response" % (k, d.title, ))
-
+            self.assertContains(r, d.title, msg_prefix="%s '%s' not in response" % (k, d.title, ))
+
+        for telechat in r.context["telechats"]:
+            # non-AD users should be 0
+            self.assertEqual(telechat["ad_pages_left_to_ballot_on"], 0, "%s ad_pages_left_to_ballot_on not in response" % (telechat["ad_pages_left_to_ballot_on"]))
+
+        username = "ad"
+        can_login = self.client.login(username=username, password="%s+password" % username)
+        self.assertTrue(can_login)
+        logged_in_request = self.client.get(url)
+        telechats = logged_in_request.context["telechats"]
+        self.assertGreater(len(telechats), 0, "Expected multiple telechats but received %d" % (len(telechats)))
+        self.assertEqual(telechats[0]["ad_pages_left_to_ballot_on"], 383, "Expected a specific number of pages left to ballot on for this AD '%s'" % (username))
```

Comment:
The returned value `telechats[0]["ad_pages_left_to_ballot_on"]` varies on every test run due to (I'm guessing) race conditions in running tests. I tried moving lines 503-510 to the `setUp` but this didn't make a reproducible value. Thoughts anyone?